### DISCUSSION TITLE
How do I change the auto numbering of a cross-ref?

### DISCUSSION THREAD

**Discussion by jbrick55** [quarto-dev/quarto-cli#9001] [Q&A] [answered]
### Description

I have a custom crossref for videos.
`crossref:
  
chapters: true  
  
fig-title: Fig
  
custom:
    - kind: float
      reference-prefix: "Video Lesson:"
      key: vid
      caption-location: top`

This works well when I have one video per page (e.g. Lesson 11.1 has Video Lesson: 11.1, Lesson 11.2 has Video Lesson: 11.2, etc..). But I would like to have the auto numbering change when multiple videos are under the same header (or page in my site). For instance, on 'Lesson 11.2' in my Quarto book, I would like to have the first video be 'Video Lesson: 11.2a' and the second video be 'Video Lesson: 11.2b' on the Lesson 11.2 page (qmd). 
Thanks for any assistance!

**Comment by mcanouil** [comment]
Could you share a small **self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.

Note that there are several other threads about this subject or related ones, did you look into them?

- https://github.com/quarto-dev/quarto-cli/discussions?discussions_q=numbering+label%3Acrossref
- https://github.com/quarto-dev/quarto-cli/issues?q=is%3Aopen+is%3Aissue+label%3Acrossref+numbering

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by mcanouil** [reply, marked as answer]
Also the custom cross-reference feature is described in <https://quarto.org/docs/authoring/cross-references-custom.html>.

> Using the custom crossref work, it's now possible to "reset" the figure numbering (by creating a new category that is rendered exactly like the original one). This covers a large number of cases.
> 
> At some point in the future we'd like to support more powerful generalization, but that requires changing more than we're comfortable doing at 1.4.

From https://github.com/quarto-dev/quarto-cli/issues/1818#issuecomment-1850716290


Globally there is no built-in way of controlling the numbering for cross-reference at this point.

**Reply by jbrick55** [reply]
Okay, that's what I was suspecting. These are helpful resources. Thanks!
